Added docs around IMemberPartialViewCacheInvalidator (#7133)
umbraco/UmbracoDocs: 16/umbraco-cms/reference/cache/imemberpartialviewcacheinvalidator.md

# ICacheRefresher

This section describes what IMemberPartialViewCacheInvalidator is, what it's default implementation does and how to customize it.

## What is an IMemberPartialViewCacheInvalidator

This interface is used to isolate the logic that needs to run to invalidate parts of the PartialView cache when a member is updated

## Why do we need to partialy invalidate the partialView cache

> Since some of the razor templates might show data that is retrieved from a member object and those templates might be cached by using the partial caching mechanism (i.e. `@await Html.CachedPartialAsync("member",Model,TimeSpan.FromDays(1), cacheByMember:true)`), we need to remove those cached partials when a member is updated.

Reviewer: ```suggestion
Razor templates may show data that is retrieved from a member object. Those templates might be cached by using the partial caching mechanism (for example, `@await Html.CachedPartialAsync("member",Model,TimeSpan.FromDays(1), cacheByMember:true)`). When a member is updated, these cached partials must be invalidated to ensure updated data is shown.
```

## Where is it used

This interface is called from the MemberCacheRefresher which is called every time a member is updated.

## Details of the implementation

> When a razor template partial is cached trough `Html.CachedPartialAsync` and `cacheByMember` is set to `true`, the extension method will append the memberId of the currently logged in member and a marker (i.e. `-m1015-`) to the partialView chachekey.

Reviewer: ```suggestion
When a razor template partial is cached through `Html.CachedPartialAsync` with `cacheByMember` set to `true`, the extension method modifies the cache key. It appends the member ID of the currently logged-in member and a marker (for example, `-m1015-`) to the partial view cache key.
```

When the `ClearPartialViewCacheItems` method is called it will clear all PartialView cacheItems that have the memberId marker for all passed in members.

> Since it is possible to call the `Html.CachedPartialAsync` with `cacheByMember` set to `true` while there is no member logged in, it will also clear all cache items with an empty member marker (i.e. `-m-`)

Reviewer: ```suggestion
If no member is logged in during caching, items with an empty member marker (for example, `-m-`) are also cleared.
```

This section describes the `IMemberPartialViewCacheInvalidator` interface, what it's default implementation does and how to customize it.

## What is an IMemberPartialViewCacheInvalidator?

This interface is used to isolate the logic that invalidates parts of the partial view cache when a member is updated.

## Why do we need to partially invalidate the partial view cache?

Razor templates may show data that is retrieved from a member object. Those templates might be cached by using the partial caching mechanism (for example, `@await Html.CachedPartialAsync("member",Model,TimeSpan.FromDays(1), cacheByMember:true)`). When a member is updated, these cached partials must be invalidated to ensure updated data is shown.

## Where is it used?

This interface is called from the member cache refresher (`MemberCacheRefresher`), which is invoked every time a member is updated.

## Details of the default implementation

Razor template partials are cached through a call to `Html.CachedPartialAsync` with `cacheByMember` set to `true`. This will append the ID of the currently logged-in member with a marker to the partial view cache key.  For example, `-m1015-`. 

> When the `ClearPartialViewCacheItems` method is called it will clear all cache items that match the marker for the updated members.

Reviewer: ```suggestion

When the `ClearPartialViewCacheItems` method is called, it will clear all cache items that match the marker for the updated members.
```
